cf_xarray/parametric.py

```diff
@@ -0,0 +1,864 @@
+    computed value is returned.
+
+    Parameters
+    ----------
+    zlev : dict
+        Attributes for `zlev` variable.
+    eta : dict
+        Attributes for `eta` variable.
+    depth : dict
+        Attributes for `depth` variable.
+
+    Returns
+    -------
+    str
+        Standard name for the computer value.
+
+    Raises
+    ------
+    ValueError
+        If `kwargs` is empty, missing values for `kwargs` keys, or could not derive the standard name.
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#table-computed-standard-names
+    """
+    found_stdname = None
+
+    allowed_names = {"zlev", "eta", "depth"}
+
+    if len(kwargs) == 0 or not (set(kwargs) <= allowed_names):
+        raise ValueError(
+            f"Must provide atleast one of {', '.join(sorted(allowed_names))}."
+        )
+
+    search_term = ""
+
+    for x, y in sorted(kwargs.items(), key=lambda x: x[0]):
+        try:
+            search_term = f"{search_term}{y['standard_name']}"
+        except TypeError:
+            raise ValueError(
+                f"The values for {', '.join(sorted(kwargs.keys()))} cannot be `None`."
+            ) from None
+        except KeyError:
+            raise ValueError(
+                f"The standard name for the {x!r} variable is not available."
+            ) from None
+
+    for x, y in ocean_stdname_map.items():
+        check_term = "".join(
+            [
+                y[i]
+                for i, j in sorted(kwargs.items(), key=lambda x: x[0])
+                if j is not None
+            ]
+        )
+
+        if search_term == check_term:
+            found_stdname = x
+
+            break
+
+    if found_stdname is None:
+        stdnames = ", ".join(
+            [y["standard_name"] for _, y in sorted(kwargs.items(), key=lambda x: x[0])]
+        )
+
+        raise ValueError(
+            f"Could not derive standard name from combination of {stdnames}."
+        )
+
+    return found_stdname
+
+
+class ParamerticVerticalCoordinate(ABC):
+    @classmethod
+    @abstractmethod
+    def from_terms(cls, terms: dict):
+        pass
+
+    @abstractmethod
+    def decode(self):
+        pass
+
+    @property
+    @abstractmethod
+    def computed_standard_name(self):
+        pass
+
+
+class AtmosphereLnPressure(ParamerticVerticalCoordinate):
+    """Atmosphere natural log pressure coordinate.
+
+    Standard name: atmosphere_ln_pressure_coordinate
+
+    Parameters
+    ----------
+    p0 : xr.DataArray
+        Reference pressure.
+    lev : xr.DataArray
+        Vertical dimensionless coordinate.
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#atmosphere-natural-log-pressure-coordinate
+    """
+
+    def __init__(self, p0, lev):
+        self.p0 = p0
+        self.lev = lev
```

Comment:
```suggestion
@dataclass
class AtmosphereLnPressure(ParamerticVerticalCoordinate):
    """Atmosphere natural log pressure coordinate.

    Standard name: atmosphere_ln_pressure_coordinate

    Parameters
    ----------
    p0 : xr.DataArray
        Reference pressure.
    lev : xr.DataArray
        Vertical dimensionless coordinate.

    References
    ----------
    Please refer to the CF conventions document :
      1. https://cfconventions.org/cf-conventions/cf-conventions.html#atmosphere-natural-log-pressure-coordinate
    """

	p0: ???
	lev: ???
```

and so on for the rest please.